deployment/migration-assistant-solution/lib/eks-infra.ts

```diff
@@ -0,0 +1,205 @@
+
+        const ec2NodeRole = new Role(this, 'MigrationsEKSEC2NodeRole', {
+            assumedBy: new ServicePrincipal('ec2.amazonaws.com'),
+            managedPolicies: [
+                ManagedPolicy.fromAwsManagedPolicyName('AmazonEKSWorkerNodePolicy'),
+                ManagedPolicy.fromAwsManagedPolicyName('AmazonEC2ContainerRegistryReadOnly'),
+                ManagedPolicy.fromAwsManagedPolicyName('AmazonEKS_CNI_Policy'),
+            ],
+        });
+
+        const subnetIds = []
+        for (const subnet of props.vpc.privateSubnets) {
+            Tags.of(subnet).add(`kubernetes.io/cluster/${props.clusterName}`, 'shared');
+            Tags.of(subnet).add('kubernetes.io/role/internal-elb', '1');
+            subnetIds.push(subnet.subnetId)
+        }
+        this.cluster = new CfnCluster(this, 'MigrationsEKSCluster', {
+            name: props.clusterName,
+            version: '1.32',
+            roleArn: clusterRole.roleArn,
+            resourcesVpcConfig: {
+                subnetIds: subnetIds,
+                endpointPrivateAccess: true,
+                endpointPublicAccess: true,
+                securityGroupIds: [migrationSecurityGroup.securityGroupId]
+            },
+            accessConfig: {
+                authenticationMode: 'API',
+            },
+            computeConfig: {
+                enabled: true,
+                nodeRoleArn: ec2NodeRole.roleArn,
+                nodePools: ["general-purpose", "system"]
+            },
+            storageConfig: {
+                blockStorage: {
+                    enabled: true
+                }
+            },
+            kubernetesNetworkConfig: {
+                elasticLoadBalancing: {
+                    enabled: true
+                }
+            }
+        });
+        migrationSecurityGroup.addIngressRule(
+            SecurityGroup.fromSecurityGroupId(this, "MigrationsEKSClusterDefaultSG", this.cluster.attrClusterSecurityGroupId),
+            Port.allTraffic()
+        );
+
+        const podIdentityRole = new CfnRole(this, 'MigrationsPodIdentityRole', {
+            roleName: `${props.clusterName}-migrations-role`,
+            assumeRolePolicyDocument: {
+                Version: '2012-10-17',
+                Statement: [
+                    {
+                        Effect: 'Allow',
+                        Principal: {
+                            Service: 'pods.eks.amazonaws.com',
+                        },
+                        Action: [
+                            "sts:AssumeRole",
+                            "sts:TagSession"
+                        ]
+                    },
+                ],
+            },
+            managedPolicyArns: [
+                ManagedPolicy.fromAwsManagedPolicyName('AmazonEC2ContainerRegistryFullAccess').managedPolicyArn,
+            ],
+            description: 'Migrations IAM role assumed by pods via EKS Pod Identity',
+        });
+
+        new CfnPolicy(this, 'MigrationsPodIdentityPolicy', {
```

Comment:
Looks like we are using the low level policy resource in CFN, can we switch to `Policy` , `PolicyStatements`, `Role`.  This will let us write them with policy `attachToRole(...)` & `addStatements(...)`

Nit: It would be cleaner if we had separate policy statements that were named in a way that indicated what they are used to access even if it is for our own knowledge.